pkg/commands/kustomizationfile.go

```diff
@@ -30,9 +34,18 @@ import (
 	"github.com/kubernetes-sigs/kustomize/pkg/types"
 )
 
+var kustomizationFields = []string{"resources", "bases", "namePrefix", "namespace", "crds", "commonLabels", "commonAnnotations", "patches", "configMapGenerator", "secretGenerator", "vars", "imageTags"}
+
+type fieldComment struct {
+	field   string
+	comment []byte
+}
+
 type kustomizationFile struct {
-	path string
-	fsys fs.FileSystem
+	path     string
+	fsys     fs.FileSystem
+	comments []fieldComment
```

Comment:
Please call this `originalFields`

Reply:
or something.  i.e. these fields were present in the original file, as opposed to ones added later.